src/native/fs/fs_napi.cpp

```diff
@@ -919,8 +918,8 @@ struct DirReadEntry : public FSWrapWorker<DirWrap>
                 } else {
                     _eof = true;
                 }
-                break;
             }
+            break;
```

Comment:
since this loop is only needed for the . and .. cases, how about we try to change it a bit like this:

```cpp
        struct dirent* e = 0;

        // loop to skip parent and current directories
        do {
            // need to set errno before the call to readdir() to detect between EOF and error
            errno = 0;
            e = readdir(dir);
        } while (e && (strcmp(e->d_name, ".") == 0 || strcmp(e->d_name, "..") == 0));

        if (e) {
            _entry.name = std::string(e->d_name);
            _entry.ino = e->d_ino;
            _entry.type = e->d_type;
        } else {
            if (errno) {
                SetSyscallError();
            } else {
                _eof = true;
            }
        }
```

Reply:
was thinking about do while :) sure will use that

Reply:
there are so little cases where do-while really helps that if you find one you have to use it :)